src/detect.ts

```diff
@@ -75,7 +43,12 @@ export async function detect({ autoInstall, programmatic, cwd }: DetectOptions =
         process.exit(1)
     }
 
-    await ezspawn(`npm i -g ${agent.split('@')[0]}${version ? `@${version}` : ''}`, { stdio: 'inherit', cwd })
+    await x(`npm i -g ${agent.split('@')[0]}${version ? `@${version}` : ''}`, undefined, {
```

Comment:
should we use `arguments` here (second argument in `x` call)?

Reply:
You should. Same as how child process works

`x('npm', ['i', '-g', ...])`

Reply:
Also keep in mind this can throw if somehow the command fails (like if `npm` doesn't exist) 
Not sure how that was being dealt with before. Maybe it's fine as is and something catches it further up